src/Components/Base/Badge/style.ts

```diff
@@ -0,0 +1,36 @@
+import styled from 'styled-components';
+import { StyledBadgeProp } from './type';
+
+const StyledBadge = styled.span<StyledBadgeProp>`
+  position: absolute;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  padding: 0.1rem 0.2rem;
+  font-size: ${({ $textSize }) => $textSize};
+
+  ${({ $position }) =>
+    ({
+      leftTop: 'top: 0px; left: 0px;',
+      leftBottom: 'bottom: 0px; left: 0px;',
+      rightTop: 'top: 0px; right: 0px;',
+      rightBottom: 'bottom: 0px; right: 0px;',
+    })[$position]};
+
+  ${({ $shape, $size }) =>
+    ({
+      circle: `
+        width: ${$size};
+        height: ${$size};
+        border-radius: 50%;`,
+      ellipse: `
+        width: auto;
+        height: auto;
+        border-radius: 45%;`,
```

Comment:
필수 x 선택 o size를 number로 받는다는 가정 o

같은 size여도 circle과 ellipse이 실제 width, height가 차이가 나서(ellipse는 비율이 auto여서 그런지 같은 size를 줘도 circle보다 높이가 더 작아지네요), 이 부분도 수정해주면 조금 더 일관적인 사이즈로 나올 것 같습니다.

```suggestion
      ellipse: `
        width: ${$size * 1.3}rem;
        height: ${$size}rem;
        border-radius: 45%;`,
```

어차피 ellipse는 children으로 받는 알림의 수를 계산해서 100이상일 땐 항상 99+로 일정한 텍스트를 가지므로 auto가 아니라 이런식으로 지정해줘도 좋을 것 같습니다.
대신, Badge의 children으로 number만 받는게 아니라, 3자 이상의 문자열도 받을 예정이라면 width는 그대로 auto로 설정해야 할 것 같습니다. 근데 저희가 아마 Badge에 문자열을 넣을 일은 딱히 없을 것 같네유.... 

이것도 개인적인 의견이니 넘기셔도 됩니다 ㅎㅎ

Reply:
적용해보겠습니다! 피드백 감사드립니다 🙇🏻‍♂️